src/Microsoft.PowerShell.GlobalTool.Shim/GlobalToolShim.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+
+namespace Microsoft.PowerShell.GlobalTool.Shim
+{
+    /// <summary>
+    /// Shim layer to chose the appropriate runtime for PowerShell Core DotNet Global tool.
+    /// </summary>
+    public class EntryPoint
+    {
+        private const string PwshDllName = "pwsh.dll";
+
+        private const string WinFolderName = "win";
+
+        private const string UnixFolderName = "unix";
+
+        /// <summary>
+        /// Entry point for the global tool.
+        /// </summary>
+        /// <param name="args">Arguments passed to the global tool.</param>
+        public static void Main(string[] args)
+        {
+            var currentPath = new FileInfo(System.Reflection.Assembly.GetEntryAssembly().Location).Directory.FullName;
+            var isWindows = Environment.OSVersion.Platform == System.PlatformID.Win32NT;
+
+            string platformFolder = isWindows ? WinFolderName : UnixFolderName;
+
+            string argsString = args.Length > 0 ? string.Join(" ", args) : null;
+            var pwshPath = Path.Combine(currentPath, platformFolder, PwshDllName);
+            string processArgs = string.IsNullOrEmpty(argsString) ? $"{pwshPath}" : $"{pwshPath} -c {argsString}";
+
+            if (File.Exists(pwshPath))
+            {
+                System.Diagnostics.Process.Start("dotnet", processArgs).WaitForExit();
+            }
+            else
+            {
+                throw new FileNotFoundException(pwshPath);
```

Comment:
Do we get really more useful exception message? Maybe not check File.Exist?

Reply:
We want to convey that the file was not found. The error message by `FileNotFoundException` is sufficient. Something similar to:

```Powershell
PS> Write-Error -Exception $q
Write-Error -Exception $q : C:\doesnotexist
+ CategoryInfo          : NotSpecified: (:) [Write-Error], FileNotFoundException
+ FullyQualifiedErrorId : System.IO.FileNotFoundException
```

We need not add resource file if the default exception is ok.

Reply:
My question was about:
```
> dotnet c:\rrrrrrr.dll
No executable found matching command "dotnet-c:\rrrrrrr.dll"
```

Reply:
I would not call `Process.Start` if the `pwshPath` does not exist.